lib/api/objectPost.js

```diff
@@ -0,0 +1,137 @@
+const async = require('async');
+const { errors, versioning } = require('arsenal');
+const { PassThrough } = require('stream');
+
+const collectCorsHeaders = require('../utilities/collectCorsHeaders');
+const createAndStoreObject = require('./apiUtils/object/createAndStoreObject');
+const { standardMetadataValidateBucketAndObj } = require('../metadata/metadataUtils');
+const { config } = require('../Config');
+const { setExpirationHeaders } = require('./apiUtils/object/expirationHeaders');
+const writeContinue = require('../utilities/writeContinue');
+const { overheadField } = require('../../constants');
+
+
+const versionIdUtils = versioning.VersionID;
+
+
+/**
+ * POST Object in the requested bucket. Steps include:
+ * validating metadata for authorization, bucket and object existence etc.
+ * store object data in datastore upon successful authorization
+ * store object location returned by datastore and
+ * object's (custom) headers in metadata
+ * return the result in final callback
+ *
+ * @param {AuthInfo} authInfo - Instance of AuthInfo class with requester's info
+ * @param {request} request - request object given by router,
+ *                            includes normalized headers
+ * @param {object | undefined } streamingV4Params - if v4 auth,
+ * object containing accessKey, signatureFromRequest, region, scopeDate,
+ * timestamp, and credentialScope
+ * (to be used for streaming v4 auth if applicable)
+ * @param {object} log - the log request
+ * @param {Function} callback - final callback to call with the result
+ * @return {undefined}
+ */
+function objectPost(authInfo, request, streamingV4Params, log, callback) {
```

Comment:
We should also have some monitoring inside this function like it's done in others

Reply:
I am not sure what monitoring means here, but indeed at the end of the day we need these calls to be logged so that we can display the new method and S3 action in our Grafana dashboards. If you want to skip that for now I am OK but in this case please create a task within the epic to make sure we do it before releasing the feature.

Reply:
In that case I'll go ahead and add the monitoring here. Thanks

For reference @davidtencer , I think @BourgoisMickael  means metric pushes like [this](https://github.com/scality/cloudserver/blob/development/8.8/lib/api/objectPut.js#L242)

Reply:
What I'm seeing from PUT Object is that it pushes promMetrics with an 'ingestSize' variable calculated from custom headers... I left it out for the moment as these are not part of my implementation as of yet. 
Should I proceed anyway?

Reply:
You can make a ticket to handle monitoring later :+1:

Reply:
I mentioned this conversation in the ticket that will handle future parrameters
https://scality.atlassian.net/jira/software/c/projects/OS/boards/216?assignee=61360619c425a20068e9dfcb&selectedIssue=S3C-8986